Pull request za merge linkerja v glavni repozitorij

thenejcar opened the pull request:
Avtomatski merge je možen, mislim da se ne bi smelo nič pokvariti. 

V README.md sem dodal komande za uporabo linkerja, pa tudi v Makefile sem dodal potrebne ukaze (na enak nacin kot za assembler). Jar datoteko sem poskusil uporabiti in mi vse deluje kot je treba.

Če najdete še kaj, kar je treba popraviti mi pa napišite kakšen mail.

lp, Nejc

jurem commented:
Živjo,

hvala za tole!

Trenutno sem si malo pogledal teste, pa bi te prosil, če bi factorial example prestavil v SicDemos. In če mu daš ime npr. link-factorial, da se bo vedelo, da je to v bistvu demo za linkanje. Razmišljam tudi, če bi komentarje v kodi prevedel v angleščino - če imaš voljo in čas.

Pa še v README.md dodaj zapis

Link - Factorial

KRATEK OPIS - ANGLEŠKI

  *   Source code: ASM datoteke
  *   Command line: morda dodaš tukaj command line, ki ga rabimo za linkanje
  *   Textual screen: 80x25 at 0B800 … screen če ga potrebuje
  *   Frequency: 10000 Hz = 10 kHz … priporočena frekvenca
  *   Author: Nejc Kišek

Hvala in lp
Jure


On 15 Sep 2016, at 8:35 , Nejc Kisek <notifications@github.com<mailto:notifications@github.com>> wrote:


Avtomatski merge je možen, mislim da se ne bi smelo nič pokvariti.

V README.md sem dodal komande za uporabo linkerja, pa tudi v Makefile sem dodal potrebne ukaze (na enak nacin kot za assembler). Jar datoteko sem poskusil uporabiti in mi vse deluje kot je treba.

Če najdete še kaj, kar je treba popraviti mi pa napišite kakšen mail.

lp, Nejc

________________________________
You can view, comment on, or merge this pull request online at:

  https://github.com/jurem/SicTools/pull/3

Commit Summary

  *   basic linking - no output yet
  *   Merge branch 'master' into linker
  *   output working, commandline options added
  *   added tests for 2 included examples, added -m -v options, better error handling
  *   added examples and tests, added force option & keep option, fixed some bugs
  *   added more error checking, example recursive factorial added
  *   added a test for factorial example
  *   linker gui added, new dropdown option in simulator (Link & load objs)
  *   added options and help to gui, fixed test paths
  *   better verbose logging, added linker to SicTools makefile and README
  *   removed output files from git
  *   fixed each Drecord in its own row - now writing 6 Drecords per row
  *   gui fixes, added a constructor for Options
  *   fixed unneccessary newlines in output obj, divided link() in 2 steps
  *   added option to edit the sections before linking
  *   gui for editing sections
  *   separated editing D and R records, deleting mrecords along with refs
  *   locil refs in defs v section gui, popravek pri enakih imenih simbolov
  *   added help to section editor
  *   correctly handle Mrecords when removing/renaming symbols, cleaned Sections class
  *   fixed a bug when renaming a reference, reorganized ui classes
  *   Merge branch 'linker'
  *   fixed missing M records in output file, also fixed tests
  *   leave +progname in M records to allow further linking
  *   fix for regular MRecords, fixed a bug in gui, added example & test (tests/linker/nametest/)
  *   removed +progname M records from output
  *   removed unused imports
  *   small fixes here and there, added one more example&test
  *   changed capitals in some methods (getTRecord, getMRecords, ...)
  *   slovenian comments in factorial example

File Changes

  *   M .gitignore<https://github.com/jurem/SicTools/pull/3/files#diff-0> (34)
  *   M Makefile<https://github.com/jurem/SicTools/pull/3/files#diff-1> (7)
  *   M README.md<https://github.com/jurem/SicTools/pull/3/files#diff-2> (23)
  *   A src/sic/Link.java<https://github.com/jurem/SicTools/pull/3/files#diff-3> (60)
  *   A src/sic/link/Linker.java<https://github.com/jurem/SicTools/pull/3/files#diff-4> (149)
  *   A src/sic/link/LinkerError.java<https://github.com/jurem/SicTools/pull/3/files#diff-5> (25)
  *   A src/sic/link/Options.java<https://github.com/jurem/SicTools/pull/3/files#diff-6> (237)
  *   A src/sic/link/section/ERecord.java<https://github.com/jurem/SicTools/pull/3/files#diff-7> (41)
  *   A src/sic/link/section/ExtDef.java<https://github.com/jurem/SicTools/pull/3/files#diff-8> (82)
  *   A src/sic/link/section/ExtRef.java<https://github.com/jurem/SicTools/pull/3/files#diff-9> (44)
  *   A src/sic/link/section/Location.java<https://github.com/jurem/SicTools/pull/3/files#diff-10> (28)
  *   A src/sic/link/section/MRecord.java<https://github.com/jurem/SicTools/pull/3/files#diff-11> (84)
  *   A src/sic/link/section/Section.java<https://github.com/jurem/SicTools/pull/3/files#diff-12> (210)
  *   A src/sic/link/section/Sections.java<https://github.com/jurem/SicTools/pull/3/files#diff-13> (333)
  *   A src/sic/link/section/TRecord.java<https://github.com/jurem/SicTools/pull/3/files#diff-14> (71)
  *   A src/sic/link/ui/EditSectionGui.java<https://github.com/jurem/SicTools/pull/3/files#diff-15> (424)
  *   A src/sic/link/ui/LinkListener.java<https://github.com/jurem/SicTools/pull/3/files#diff-16> (8)
  *   A src/sic/link/ui/LinkerCli.java<https://github.com/jurem/SicTools/pull/3/files#diff-17> (344)
  *   A src/sic/link/ui/LinkerGui.java<https://github.com/jurem/SicTools/pull/3/files#diff-18> (450)
  *   A src/sic/link/ui/SectionEditListener.java<https://github.com/jurem/SicTools/pull/3/files#diff-19> (8)
  *   A src/sic/link/utils/Parser.java<https://github.com/jurem/SicTools/pull/3/files#diff-20> (222)
  *   A src/sic/link/utils/Writer.java<https://github.com/jurem/SicTools/pull/3/files#diff-21> (126)
  *   A src/sic/link/visitors/FirstPassVisitor.java<https://github.com/jurem/SicTools/pull/3/files#diff-22> (102)
  *   A src/sic/link/visitors/SecondPassVisitor.java<https://github.com/jurem/SicTools/pull/3/files#diff-23> (106)
  *   A src/sic/link/visitors/SectionVisitor.java<https://github.com/jurem/SicTools/pull/3/files#diff-24> (46)
  *   M src/sic/sim/MainView.java<https://github.com/jurem/SicTools/pull/3/files#diff-25> (23)
  *   A tests/linker/LinkerTest.java<https://github.com/jurem/SicTools/pull/3/files#diff-26> (759)
  *   A tests/linker/absolute/abs.asm<https://github.com/jurem/SicTools/pull/3/files#diff-27> (13)
  *   A tests/linker/absolute/abs.obj<https://github.com/jurem/SicTools/pull/3/files#diff-28> (4)
  *   A tests/linker/absolute/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-29> (5)
  *   A tests/linker/absolute/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-30> (5)
  *   A tests/linker/factorial/ending.asm<https://github.com/jurem/SicTools/pull/3/files#diff-31> (4)
  *   A tests/linker/factorial/ending.obj<https://github.com/jurem/SicTools/pull/3/files#diff-32> (4)
  *   A tests/linker/factorial/fact.asm<https://github.com/jurem/SicTools/pull/3/files#diff-33> (32)
  *   A tests/linker/factorial/fact.obj<https://github.com/jurem/SicTools/pull/3/files#diff-34> (11)
  *   A tests/linker/factorial/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-35> (24)
  *   A tests/linker/factorial/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-36> (11)
  *   A tests/linker/factorial/print.asm<https://github.com/jurem/SicTools/pull/3/files#diff-37> (36)
  *   A tests/linker/factorial/print.obj<https://github.com/jurem/SicTools/pull/3/files#diff-38> (6)
  *   A tests/linker/factorial/print2.obj<https://github.com/jurem/SicTools/pull/3/files#diff-39> (8)
  *   A tests/linker/factorial/stack.asm<https://github.com/jurem/SicTools/pull/3/files#diff-40> (19)
  *   A tests/linker/factorial/stack.obj<https://github.com/jurem/SicTools/pull/3/files#diff-41> (5)
  *   A tests/linker/multi/func.asm<https://github.com/jurem/SicTools/pull/3/files#diff-42> (38)
  *   A tests/linker/multi/func.obj<https://github.com/jurem/SicTools/pull/3/files#diff-43> (22)
  *   A tests/linker/multi/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-44> (5)
  *   A tests/linker/multi/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-45> (5)
  *   A tests/linker/nametest/extd.asm<https://github.com/jurem/SicTools/pull/3/files#diff-46> (5)
  *   A tests/linker/nametest/extd.obj<https://github.com/jurem/SicTools/pull/3/files#diff-47> (4)
  *   A tests/linker/nametest/first.asm<https://github.com/jurem/SicTools/pull/3/files#diff-48> (8)
  *   A tests/linker/nametest/first.obj<https://github.com/jurem/SicTools/pull/3/files#diff-49> (5)
  *   A tests/linker/nametest/test.asm<https://github.com/jurem/SicTools/pull/3/files#diff-50> (28)
  *   A tests/linker/nametest/test.obj<https://github.com/jurem/SicTools/pull/3/files#diff-51> (13)
  *   A tests/linker/one/file.asm<https://github.com/jurem/SicTools/pull/3/files#diff-52> (24)
  *   A tests/linker/one/file.obj<https://github.com/jurem/SicTools/pull/3/files#diff-53> (13)
  *   A tests/linker/one2/prog.asm<https://github.com/jurem/SicTools/pull/3/files#diff-54> (24)
  *   A tests/linker/one2/prog.obj<https://github.com/jurem/SicTools/pull/3/files#diff-55> (14)
  *   A tests/linker/partial/lib.asm<https://github.com/jurem/SicTools/pull/3/files#diff-56> (6)
  *   A tests/linker/partial/lib.obj<https://github.com/jurem/SicTools/pull/3/files#diff-57> (4)
  *   A tests/linker/partial/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-58> (15)
  *   A tests/linker/partial/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-59> (7)
  *   A tests/linker/stack/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-60> (34)
  *   A tests/linker/stack/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-61> (15)
  *   A tests/linker/stack/stack.asm<https://github.com/jurem/SicTools/pull/3/files#diff-62> (25)
  *   A tests/linker/stack/stack.obj<https://github.com/jurem/SicTools/pull/3/files#diff-63> (5)

Patch Links:

  *   https://github.com/jurem/SicTools/pull/3.patch
  *   https://github.com/jurem/SicTools/pull/3.diff

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/jurem/SicTools/pull/3>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ABxG7-bzNHsb0M_WKLS2T3ZEQ9kC3Auaks5qqOckgaJpZM4J9im9>.

thenejcar commented:
Zivjo,

bom danes dodal tisti primer v SicDemos in zraven dal še eno skripto/navodila za povezovanje. Angleško verzijo komentarjev pa že imam nekje — v slovenščino sem jih prevedel, da sem dal kodo kot primer v diplomo, originalno sem jih napisal pa v angleščini. 

lp, Nejc


> On 17 Sep 2016, at 08:54, Jure Mihelič <notifications@github.com> wrote:
> 
> Živjo, 
> 
> hvala za tole! 
> 
> Trenutno sem si malo pogledal teste, pa bi te prosil, če bi factorial example prestavil v SicDemos. In če mu daš ime npr. link-factorial, da se bo vedelo, da je to v bistvu demo za linkanje. Razmišljam tudi, če bi komentarje v kodi prevedel v angleščino - če imaš voljo in čas. 
> 
> Pa še v README.md dodaj zapis 
> 
> Link - Factorial 
> 
> KRATEK OPIS - ANGLEŠKI 
> 
> * Source code: ASM datoteke 
> * Command line: morda dodaš tukaj command line, ki ga rabimo za linkanje 
> * Textual screen: 80x25 at 0B800 … screen če ga potrebuje 
> * Frequency: 10000 Hz = 10 kHz … priporočena frekvenca 
> * Author: Nejc Kišek 
> 
> Hvala in lp 
> Jure 
> 
> 
> On 15 Sep 2016, at 8:35 , Nejc Kisek <notifications@github.com<mailto:notifications@github.com>> wrote: 
> 
> 
> Avtomatski merge je možen, mislim da se ne bi smelo nič pokvariti. 
> 
> V README.md sem dodal komande za uporabo linkerja, pa tudi v Makefile sem dodal potrebne ukaze (na enak nacin kot za assembler). Jar datoteko sem poskusil uporabiti in mi vse deluje kot je treba. 
> 
> Če najdete še kaj, kar je treba popraviti mi pa napišite kakšen mail. 
> 
> lp, Nejc 
> 
> ________________________________ 
> You can view, comment on, or merge this pull request online at: 
> 
> https://github.com/jurem/SicTools/pull/3 
> 
> Commit Summary 
> 
> * basic linking - no output yet 
> * Merge branch 'master' into linker 
> * output working, commandline options added 
> * added tests for 2 included examples, added -m -v options, better error handling 
> * added examples and tests, added force option & keep option, fixed some bugs 
> * added more error checking, example recursive factorial added 
> * added a test for factorial example 
> * linker gui added, new dropdown option in simulator (Link & load objs) 
> * added options and help to gui, fixed test paths 
> * better verbose logging, added linker to SicTools makefile and README 
> * removed output files from git 
> * fixed each Drecord in its own row - now writing 6 Drecords per row 
> * gui fixes, added a constructor for Options 
> * fixed unneccessary newlines in output obj, divided link() in 2 steps 
> * added option to edit the sections before linking 
> * gui for editing sections 
> * separated editing D and R records, deleting mrecords along with refs 
> * locil refs in defs v section gui, popravek pri enakih imenih simbolov 
> * added help to section editor 
> * correctly handle Mrecords when removing/renaming symbols, cleaned Sections class 
> * fixed a bug when renaming a reference, reorganized ui classes 
> * Merge branch 'linker' 
> * fixed missing M records in output file, also fixed tests 
> * leave +progname in M records to allow further linking 
> * fix for regular MRecords, fixed a bug in gui, added example & test (tests/linker/nametest/) 
> * removed +progname M records from output 
> * removed unused imports 
> * small fixes here and there, added one more example&test 
> * changed capitals in some methods (getTRecord, getMRecords, ...) 
> * slovenian comments in factorial example 
> 
> File Changes 
> 
> * M .gitignore<https://github.com/jurem/SicTools/pull/3/files#diff-0> (34) 
> * M Makefile<https://github.com/jurem/SicTools/pull/3/files#diff-1> (7) 
> * M README.md<https://github.com/jurem/SicTools/pull/3/files#diff-2> (23) 
> * A src/sic/Link.java<https://github.com/jurem/SicTools/pull/3/files#diff-3> (60) 
> * A src/sic/link/Linker.java<https://github.com/jurem/SicTools/pull/3/files#diff-4> (149) 
> * A src/sic/link/LinkerError.java<https://github.com/jurem/SicTools/pull/3/files#diff-5> (25) 
> * A src/sic/link/Options.java<https://github.com/jurem/SicTools/pull/3/files#diff-6> (237) 
> * A src/sic/link/section/ERecord.java<https://github.com/jurem/SicTools/pull/3/files#diff-7> (41) 
> * A src/sic/link/section/ExtDef.java<https://github.com/jurem/SicTools/pull/3/files#diff-8> (82) 
> * A src/sic/link/section/ExtRef.java<https://github.com/jurem/SicTools/pull/3/files#diff-9> (44) 
> * A src/sic/link/section/Location.java<https://github.com/jurem/SicTools/pull/3/files#diff-10> (28) 
> * A src/sic/link/section/MRecord.java<https://github.com/jurem/SicTools/pull/3/files#diff-11> (84) 
> * A src/sic/link/section/Section.java<https://github.com/jurem/SicTools/pull/3/files#diff-12> (210) 
> * A src/sic/link/section/Sections.java<https://github.com/jurem/SicTools/pull/3/files#diff-13> (333) 
> * A src/sic/link/section/TRecord.java<https://github.com/jurem/SicTools/pull/3/files#diff-14> (71) 
> * A src/sic/link/ui/EditSectionGui.java<https://github.com/jurem/SicTools/pull/3/files#diff-15> (424) 
> * A src/sic/link/ui/LinkListener.java<https://github.com/jurem/SicTools/pull/3/files#diff-16> (8) 
> * A src/sic/link/ui/LinkerCli.java<https://github.com/jurem/SicTools/pull/3/files#diff-17> (344) 
> * A src/sic/link/ui/LinkerGui.java<https://github.com/jurem/SicTools/pull/3/files#diff-18> (450) 
> * A src/sic/link/ui/SectionEditListener.java<https://github.com/jurem/SicTools/pull/3/files#diff-19> (8) 
> * A src/sic/link/utils/Parser.java<https://github.com/jurem/SicTools/pull/3/files#diff-20> (222) 
> * A src/sic/link/utils/Writer.java<https://github.com/jurem/SicTools/pull/3/files#diff-21> (126) 
> * A src/sic/link/visitors/FirstPassVisitor.java<https://github.com/jurem/SicTools/pull/3/files#diff-22> (102) 
> * A src/sic/link/visitors/SecondPassVisitor.java<https://github.com/jurem/SicTools/pull/3/files#diff-23> (106) 
> * A src/sic/link/visitors/SectionVisitor.java<https://github.com/jurem/SicTools/pull/3/files#diff-24> (46) 
> * M src/sic/sim/MainView.java<https://github.com/jurem/SicTools/pull/3/files#diff-25> (23) 
> * A tests/linker/LinkerTest.java<https://github.com/jurem/SicTools/pull/3/files#diff-26> (759) 
> * A tests/linker/absolute/abs.asm<https://github.com/jurem/SicTools/pull/3/files#diff-27> (13) 
> * A tests/linker/absolute/abs.obj<https://github.com/jurem/SicTools/pull/3/files#diff-28> (4) 
> * A tests/linker/absolute/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-29> (5) 
> * A tests/linker/absolute/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-30> (5) 
> * A tests/linker/factorial/ending.asm<https://github.com/jurem/SicTools/pull/3/files#diff-31> (4) 
> * A tests/linker/factorial/ending.obj<https://github.com/jurem/SicTools/pull/3/files#diff-32> (4) 
> * A tests/linker/factorial/fact.asm<https://github.com/jurem/SicTools/pull/3/files#diff-33> (32) 
> * A tests/linker/factorial/fact.obj<https://github.com/jurem/SicTools/pull/3/files#diff-34> (11) 
> * A tests/linker/factorial/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-35> (24) 
> * A tests/linker/factorial/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-36> (11) 
> * A tests/linker/factorial/print.asm<https://github.com/jurem/SicTools/pull/3/files#diff-37> (36) 
> * A tests/linker/factorial/print.obj<https://github.com/jurem/SicTools/pull/3/files#diff-38> (6) 
> * A tests/linker/factorial/print2.obj<https://github.com/jurem/SicTools/pull/3/files#diff-39> (8) 
> * A tests/linker/factorial/stack.asm<https://github.com/jurem/SicTools/pull/3/files#diff-40> (19) 
> * A tests/linker/factorial/stack.obj<https://github.com/jurem/SicTools/pull/3/files#diff-41> (5) 
> * A tests/linker/multi/func.asm<https://github.com/jurem/SicTools/pull/3/files#diff-42> (38) 
> * A tests/linker/multi/func.obj<https://github.com/jurem/SicTools/pull/3/files#diff-43> (22) 
> * A tests/linker/multi/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-44> (5) 
> * A tests/linker/multi/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-45> (5) 
> * A tests/linker/nametest/extd.asm<https://github.com/jurem/SicTools/pull/3/files#diff-46> (5) 
> * A tests/linker/nametest/extd.obj<https://github.com/jurem/SicTools/pull/3/files#diff-47> (4) 
> * A tests/linker/nametest/first.asm<https://github.com/jurem/SicTools/pull/3/files#diff-48> (8) 
> * A tests/linker/nametest/first.obj<https://github.com/jurem/SicTools/pull/3/files#diff-49> (5) 
> * A tests/linker/nametest/test.asm<https://github.com/jurem/SicTools/pull/3/files#diff-50> (28) 
> * A tests/linker/nametest/test.obj<https://github.com/jurem/SicTools/pull/3/files#diff-51> (13) 
> * A tests/linker/one/file.asm<https://github.com/jurem/SicTools/pull/3/files#diff-52> (24) 
> * A tests/linker/one/file.obj<https://github.com/jurem/SicTools/pull/3/files#diff-53> (13) 
> * A tests/linker/one2/prog.asm<https://github.com/jurem/SicTools/pull/3/files#diff-54> (24) 
> * A tests/linker/one2/prog.obj<https://github.com/jurem/SicTools/pull/3/files#diff-55> (14) 
> * A tests/linker/partial/lib.asm<https://github.com/jurem/SicTools/pull/3/files#diff-56> (6) 
> * A tests/linker/partial/lib.obj<https://github.com/jurem/SicTools/pull/3/files#diff-57> (4) 
> * A tests/linker/partial/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-58> (15) 
> * A tests/linker/partial/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-59> (7) 
> * A tests/linker/stack/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-60> (34) 
> * A tests/linker/stack/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-61> (15) 
> * A tests/linker/stack/stack.asm<https://github.com/jurem/SicTools/pull/3/files#diff-62> (25) 
> * A tests/linker/stack/stack.obj<https://github.com/jurem/SicTools/pull/3/files#diff-63> (5) 
> 
> Patch Links: 
> 
> * https://github.com/jurem/SicTools/pull/3.patch 
> * https://github.com/jurem/SicTools/pull/3.diff 
> 
> — 
> You are receiving this because you are subscribed to this thread. 
> Reply to this email directly, view it on GitHub<https://github.com/jurem/SicTools/pull/3>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ABxG7-bzNHsb0M_WKLS2T3ZEQ9kC3Auaks5qqOckgaJpZM4J9im9>. 
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/jurem/SicTools/pull/3#issuecomment-247753800>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AGc1uCJ1EGAGL2_qg2OwEFjcDeejCHn9ks5qq46kgaJpZM4J9im9>.
>

sueyllam commented:
Attached is a new archive for sic-tools with Loader.java hacked to fix a problem with eoln on Windows. Also added updateview to loadobj in Mainvies.java.A small fix also was added to the factorial test case, so now it is working perfectly after linking main.obj, fact.obj, print.obj and stack.obj hisham... 

    On Saturday, September 17, 2016 10:17 AM, Nejc Kisek <notifications@github.com> wrote:
 

 Zivjo,

bom danes dodal tisti primer v SicDemos in zraven dal še eno skripto/navodila za povezovanje. Angleško verzijo komentarjev pa že imam nekje — v slovenščino sem jih prevedel, da sem dal kodo kot primer v diplomo, originalno sem jih napisal pa v angleščini. 

lp, Nejc


> On 17 Sep 2016, at 08:54, Jure Mihelič <notifications@github.com> wrote:
> 
> Živjo, 
> 
> hvala za tole! 
> 
> Trenutno sem si malo pogledal teste, pa bi te prosil, če bi factorial example prestavil v SicDemos. In če mu daš ime npr. link-factorial, da se bo vedelo, da je to v bistvu demo za linkanje. Razmišljam tudi, če bi komentarje v kodi prevedel v angleščino - če imaš voljo in čas. 
> 
> Pa še v README.md dodaj zapis 
> 
> Link - Factorial 
> 
> KRATEK OPIS - ANGLEŠKI 
> 
> * Source code: ASM datoteke 
> * Command line: morda dodaš tukaj command line, ki ga rabimo za linkanje 
> * Textual screen: 80x25 at 0B800 … screen če ga potrebuje 
> * Frequency: 10000 Hz = 10 kHz … priporočena frekvenca 
> * Author: Nejc Kišek 
> 
> Hvala in lp 
> Jure 
> 
> 
> On 15 Sep 2016, at 8:35 , Nejc Kisek <notifications@github.com<mailto:notifications@github.com>> wrote: 
> 
> 
> Avtomatski merge je možen, mislim da se ne bi smelo nič pokvariti. 
> 
> V README.md sem dodal komande za uporabo linkerja, pa tudi v Makefile sem dodal potrebne ukaze (na enak nacin kot za assembler). Jar datoteko sem poskusil uporabiti in mi vse deluje kot je treba. 
> 
> Če najdete še kaj, kar je treba popraviti mi pa napišite kakšen mail. 
> 
> lp, Nejc 
> 
> ________________________________ 
> You can view, comment on, or merge this pull request online at: 
> 
> https://github.com/jurem/SicTools/pull/3 
> 
> Commit Summary 
> 
> * basic linking - no output yet 
> * Merge branch 'master' into linker 
> * output working, commandline options added 
> * added tests for 2 included examples, added -m -v options, better error handling 
> * added examples and tests, added force option & keep option, fixed some bugs 
> * added more error checking, example recursive factorial added 
> * added a test for factorial example 
> * linker gui added, new dropdown option in simulator (Link & load objs) 
> * added options and help to gui, fixed test paths 
> * better verbose logging, added linker to SicTools makefile and README 
> * removed output files from git 
> * fixed each Drecord in its own row - now writing 6 Drecords per row 
> * gui fixes, added a constructor for Options 
> * fixed unneccessary newlines in output obj, divided link() in 2 steps 
> * added option to edit the sections before linking 
> * gui for editing sections 
> * separated editing D and R records, deleting mrecords along with refs 
> * locil refs in defs v section gui, popravek pri enakih imenih simbolov 
> * added help to section editor 
> * correctly handle Mrecords when removing/renaming symbols, cleaned Sections class 
> * fixed a bug when renaming a reference, reorganized ui classes 
> * Merge branch 'linker' 
> * fixed missing M records in output file, also fixed tests 
> * leave +progname in M records to allow further linking 
> * fix for regular MRecords, fixed a bug in gui, added example & test (tests/linker/nametest/) 
> * removed +progname M records from output 
> * removed unused imports 
> * small fixes here and there, added one more example&test 
> * changed capitals in some methods (getTRecord, getMRecords, ...) 
> * slovenian comments in factorial example 
> 
> File Changes 
> 
> * M .gitignore<https://github.com/jurem/SicTools/pull/3/files#diff-0> (34) 
> * M Makefile<https://github.com/jurem/SicTools/pull/3/files#diff-1> (7) 
> * M README.md<https://github.com/jurem/SicTools/pull/3/files#diff-2> (23) 
> * A src/sic/Link.java<https://github.com/jurem/SicTools/pull/3/files#diff-3> (60) 
> * A src/sic/link/Linker.java<https://github.com/jurem/SicTools/pull/3/files#diff-4> (149) 
> * A src/sic/link/LinkerError.java<https://github.com/jurem/SicTools/pull/3/files#diff-5> (25) 
> * A src/sic/link/Options.java<https://github.com/jurem/SicTools/pull/3/files#diff-6> (237) 
> * A src/sic/link/section/ERecord.java<https://github.com/jurem/SicTools/pull/3/files#diff-7> (41) 
> * A src/sic/link/section/ExtDef.java<https://github.com/jurem/SicTools/pull/3/files#diff-8> (82) 
> * A src/sic/link/section/ExtRef.java<https://github.com/jurem/SicTools/pull/3/files#diff-9> (44) 
> * A src/sic/link/section/Location.java<https://github.com/jurem/SicTools/pull/3/files#diff-10> (28) 
> * A src/sic/link/section/MRecord.java<https://github.com/jurem/SicTools/pull/3/files#diff-11> (84) 
> * A src/sic/link/section/Section.java<https://github.com/jurem/SicTools/pull/3/files#diff-12> (210) 
> * A src/sic/link/section/Sections.java<https://github.com/jurem/SicTools/pull/3/files#diff-13> (333) 
> * A src/sic/link/section/TRecord.java<https://github.com/jurem/SicTools/pull/3/files#diff-14> (71) 
> * A src/sic/link/ui/EditSectionGui.java<https://github.com/jurem/SicTools/pull/3/files#diff-15> (424) 
> * A src/sic/link/ui/LinkListener.java<https://github.com/jurem/SicTools/pull/3/files#diff-16> (8) 
> * A src/sic/link/ui/LinkerCli.java<https://github.com/jurem/SicTools/pull/3/files#diff-17> (344) 
> * A src/sic/link/ui/LinkerGui.java<https://github.com/jurem/SicTools/pull/3/files#diff-18> (450) 
> * A src/sic/link/ui/SectionEditListener.java<https://github.com/jurem/SicTools/pull/3/files#diff-19> (8) 
> * A src/sic/link/utils/Parser.java<https://github.com/jurem/SicTools/pull/3/files#diff-20> (222) 
> * A src/sic/link/utils/Writer.java<https://github.com/jurem/SicTools/pull/3/files#diff-21> (126) 
> * A src/sic/link/visitors/FirstPassVisitor.java<https://github.com/jurem/SicTools/pull/3/files#diff-22> (102) 
> * A src/sic/link/visitors/SecondPassVisitor.java<https://github.com/jurem/SicTools/pull/3/files#diff-23> (106) 
> * A src/sic/link/visitors/SectionVisitor.java<https://github.com/jurem/SicTools/pull/3/files#diff-24> (46) 
> * M src/sic/sim/MainView.java<https://github.com/jurem/SicTools/pull/3/files#diff-25> (23) 
> * A tests/linker/LinkerTest.java<https://github.com/jurem/SicTools/pull/3/files#diff-26> (759) 
> * A tests/linker/absolute/abs.asm<https://github.com/jurem/SicTools/pull/3/files#diff-27> (13) 
> * A tests/linker/absolute/abs.obj<https://github.com/jurem/SicTools/pull/3/files#diff-28> (4) 
> * A tests/linker/absolute/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-29> (5) 
> * A tests/linker/absolute/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-30> (5) 
> * A tests/linker/factorial/ending.asm<https://github.com/jurem/SicTools/pull/3/files#diff-31> (4) 
> * A tests/linker/factorial/ending.obj<https://github.com/jurem/SicTools/pull/3/files#diff-32> (4) 
> * A tests/linker/factorial/fact.asm<https://github.com/jurem/SicTools/pull/3/files#diff-33> (32) 
> * A tests/linker/factorial/fact.obj<https://github.com/jurem/SicTools/pull/3/files#diff-34> (11) 
> * A tests/linker/factorial/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-35> (24) 
> * A tests/linker/factorial/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-36> (11) 
> * A tests/linker/factorial/print.asm<https://github.com/jurem/SicTools/pull/3/files#diff-37> (36) 
> * A tests/linker/factorial/print.obj<https://github.com/jurem/SicTools/pull/3/files#diff-38> (6) 
> * A tests/linker/factorial/print2.obj<https://github.com/jurem/SicTools/pull/3/files#diff-39> (8) 
> * A tests/linker/factorial/stack.asm<https://github.com/jurem/SicTools/pull/3/files#diff-40> (19) 
> * A tests/linker/factorial/stack.obj<https://github.com/jurem/SicTools/pull/3/files#diff-41> (5) 
> * A tests/linker/multi/func.asm<https://github.com/jurem/SicTools/pull/3/files#diff-42> (38) 
> * A tests/linker/multi/func.obj<https://github.com/jurem/SicTools/pull/3/files#diff-43> (22) 
> * A tests/linker/multi/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-44> (5) 
> * A tests/linker/multi/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-45> (5) 
> * A tests/linker/nametest/extd.asm<https://github.com/jurem/SicTools/pull/3/files#diff-46> (5) 
> * A tests/linker/nametest/extd.obj<https://github.com/jurem/SicTools/pull/3/files#diff-47> (4) 
> * A tests/linker/nametest/first.asm<https://github.com/jurem/SicTools/pull/3/files#diff-48> (8) 
> * A tests/linker/nametest/first.obj<https://github.com/jurem/SicTools/pull/3/files#diff-49> (5) 
> * A tests/linker/nametest/test.asm<https://github.com/jurem/SicTools/pull/3/files#diff-50> (28) 
> * A tests/linker/nametest/test.obj<https://github.com/jurem/SicTools/pull/3/files#diff-51> (13) 
> * A tests/linker/one/file.asm<https://github.com/jurem/SicTools/pull/3/files#diff-52> (24) 
> * A tests/linker/one/file.obj<https://github.com/jurem/SicTools/pull/3/files#diff-53> (13) 
> * A tests/linker/one2/prog.asm<https://github.com/jurem/SicTools/pull/3/files#diff-54> (24) 
> * A tests/linker/one2/prog.obj<https://github.com/jurem/SicTools/pull/3/files#diff-55> (14) 
> * A tests/linker/partial/lib.asm<https://github.com/jurem/SicTools/pull/3/files#diff-56> (6) 
> * A tests/linker/partial/lib.obj<https://github.com/jurem/SicTools/pull/3/files#diff-57> (4) 
> * A tests/linker/partial/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-58> (15) 
> * A tests/linker/partial/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-59> (7) 
> * A tests/linker/stack/main.asm<https://github.com/jurem/SicTools/pull/3/files#diff-60> (34) 
> * A tests/linker/stack/main.obj<https://github.com/jurem/SicTools/pull/3/files#diff-61> (15) 
> * A tests/linker/stack/stack.asm<https://github.com/jurem/SicTools/pull/3/files#diff-62> (25) 
> * A tests/linker/stack/stack.obj<https://github.com/jurem/SicTools/pull/3/files#diff-63> (5) 
> 
> Patch Links: 
> 
> * https://github.com/jurem/SicTools/pull/3.patch 
> * https://github.com/jurem/SicTools/pull/3.diff 
> 
> — 
> You are receiving this because you are subscribed to this thread. 
> Reply to this email directly, view it on GitHub<https://github.com/jurem/SicTools/pull/3>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ABxG7-bzNHsb0M_WKLS2T3ZEQ9kC3Auaks5qqOckgaJpZM4J9im9>. 
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/jurem/SicTools/pull/3#issuecomment-247753800>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AGc1uCJ1EGAGL2_qg2OwEFjcDeejCHn9ks5qq46kgaJpZM4J9im9>.
> 

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or mute the thread.

jurem commented:
Sorry, cannot find any pull request on the github.

On 19. sep. 2016, at 16:57, sueyllam <notifications@github.com<mailto:notifications@github.com>> wrote:

Attached is a new archive for sic-tools with Loader.java hacked to fix a problem with eoln on Windows. Also added updateview to loadobj in Mainvies.java.A small fix also was added to the factorial test case, so now it is working perfectly after linking main.obj, fact.obj, print.obj and stack.obj hisham...

sueyllam commented:
3 pull requests were created... hisham... 

    On Wednesday, September 21, 2016 9:35 AM, Jure Mihelič <notifications@github.com> wrote:
 

 Sorry, cannot find any pull request on the github.

On 19. sep. 2016, at 16:57, sueyllam <notifications@github.com<mailto:notifications@github.com>> wrote:

Attached is a new archive for sic-tools with Loader.java hacked to fix a problem with eoln on Windows. Also added updateview to loadobj in Mainvies.java.A small fix also was added to the factorial test case, so now it is working perfectly after linking main.obj, fact.obj, print.obj and stack.obj hisham...
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.
